hs_core/hydroshare/resource.py

```diff
@@ -751,14 +750,13 @@ def publish_resource(pk):
     Note:  This is different than just giving public access to a resource via access control rul
     """
     resource = utils.get_resource_by_shortkey(pk)
-    resource.published_and_frozen = True
-    resource.frozen = True
-    resource.edit_users = []
-    resource.edit_groups = []
+    resource.raccess.published = True
```

Comment:
It looks like the original use of AbstractResource.edit_users  seems to be hacked to react to the immutable flag by emptying this list. I can emulate that in my fake list if this is desirable. Meanwhile one should be careful because AbstractResource.edit_users is hacked while GenericResource.raccess.edit_users has contents. Thus one cannot replace the first with the second.